rules-engine/tests/rules_engine/test_engine.py

```diff
@@ -41,23 +31,50 @@ def test_average_indoor_temp():
 
 
 def test_bp_ua_estimates():
-    avg_temps_1 = [[28, 29, 30, 29], [32, 35, 35, 38], [41, 43, 42, 42]]
-    usages_1 = [50, 45, 30]
-    bp, uas, avg_ua, stdev_pct = engine.bp_ua_estimates(
-        "gas", 0.24, 0.88, avg_temps_1, usages_1, initial_bp=58
+    home = engine.Home(
+        engine.FuelType.GAS, heat_sys_efficiency=0.88, initial_balance_point=58
+    )
+
+    daily_temps_lists = [[28, 29, 30, 29], [32, 35, 35, 38], [41, 43, 42, 42]]
+    usages = [50, 45, 30]
+    home.initialize_billing_periods(daily_temps_lists, usages, 0.24)
+    home.calculate_balance_point_and_ua()
+    ua_1, ua_2, ua_3 = [bill.ua for bill in home.bills]
+
+    assert home.balance_point == 60
+    assert ua_1 == approx(1450.5, abs=1)
+    assert ua_2 == approx(1615.3, abs=1)
+    assert ua_3 == approx(1479.6, abs=1)
+    assert home.avg_ua == approx(1515.1, abs=1)
+    assert home.stdev_pct == approx(0.0474, abs=0.01)
+
+
+def test_bp_ua_with_outlier():
+    home = engine.Home(
+        engine.FuelType.GAS, heat_sys_efficiency=0.88, initial_balance_point=58
     )
-    ua_1, ua_2, ua_3 = uas
-    assert bp == 60
+    daily_temps_lists = [
+        [41.7, 41.6, 32, 25.4],
+        [28, 29, 30, 29],
+        [32, 35, 35, 38],
+        [41, 43, 42, 42],
+    ]
+    usages = [60, 50, 45, 30]
+    home.initialize_billing_periods(daily_temps_lists, usages, 0.24)
+    home.calculate_balance_point_and_ua()
+    ua_1, ua_2, ua_3 = [bill.ua for bill in home.bills]
+
+    assert home.balance_point == 60
     assert ua_1 == approx(1450.5, abs=1)
     assert ua_2 == approx(1615.3, abs=1)
     assert ua_3 == approx(1479.6, abs=1)
-    assert avg_ua == approx(1515.1, abs=1)
-    assert stdev_pct == approx(0.0474, abs=0.01)
+    assert home.avg_ua == approx(1515.1, abs=1)
+    assert home.stdev_pct == approx(0.0474, abs=0.01)
 
 
 if __name__ == "__main__":
```

Comment:
`nit`: what is this being used for?  If it's for running tests, I don't think we need this given `pytest` is our test runner.  e.g. I can delete this section and run `pytest` from my `venv` and all tests are run.  Or is it used for something else?